### PR TITLE
feat: upgrade urllib3 to 1.26

### PR DESCRIPTION
**Pull request by joshuarli**
OPS-976.

Revival of https://github.com/getsentry/sentry/pull/30824 now that responses upgrade is more of a priority (cc @scefali). Also, unblocks selenium 4 upgrade, but that's also sort of on us that selenium's included in requirements-base rather than dev.

Regarding our urllib3 overrides, I've resynced with upstream and improved annotation comments to show where our custom code is.

Only thing from the changelog that might start yelling in production, which I'm fine ignoring for now since we can easily swithch it on if it leads to too much sentry error noise:

> Deprecated negotiating TLSv1 and TLSv1.1 by default. Users that still wish to use TLS earlier than 1.2 without a deprecation warning should opt-in explicitly by setting ssl_version=ssl.PROTOCOL_TLSv1_1 (Pull #2002) Starting in urllib3 v2.0: Connections that receive a ``DeprecationWarning`` will fail

Also, requires boto3 bump. The changelog is.... unreviewable to me. Blocked by https://github.com/getsentry/sentry/pull/34524.

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -66,7 +66,7 @@ toronado==0.1.0
 typing-extensions==3.10.0.2
 ua-parser==0.10.0
 unidiff==0.7.1
-urllib3[brotli]==1.25.11
+urllib3[brotli]==1.26.8
 brotlipy==0.7.0
```
Comment (asottile-sentry) [inline requirements-base.txt:70]: urllib changed the brotli library -- you'll want to make sure you're using the right one (or just remove the `brotlipy` pin)

Comment (joshuarli) [reply requirements-base.txt:70]: Good catch, I was just relying on their changelog (which doesn't document https://github.com/urllib3/urllib3/issues/2574) and didn't actually check.

File: requirements-base.txt [hunk 2]
```diff
@@ -67,8 +67,8 @@ toronado==0.1.0
 typing-extensions==3.10.0.2
 ua-parser==0.10.0
 unidiff==0.7.1
-urllib3[brotli]==1.25.11
-brotlipy==0.7.0
+urllib3[brotli]==1.26.8
```
Comment (mdtro) [inline requirements-base.txt:70]: Can we bump to 1.26.9? 😄

Comment (joshuarli) [reply requirements-base.txt:70]: Yeah. Thanks for letting me know, forgot about this. Now that https://github.com/getsentry/sentry/pull/34524 is done, we can do this.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**